packages/dds/tree/src/core/tree/anchorSet.ts

```diff
@@ -516,21 +516,27 @@ export class AnchorSet implements Listenable<AnchorSetRootEvents>, AnchorLocator
 			let index = 0;
 			while (
 				index < sourceChildren.length &&
-				sourceChildren[index].parentIndex < startPath.parentIndex
+				// Non null asserting here because of the length check above
+				// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+				sourceChildren[index]!.parentIndex < startPath.parentIndex
```

Comment:
Since all five of the violations in this file are of the same kind, I'd suggest just making a helper function in this file:

```ts
function getParentIndex(nodes: PathNode[], index: number): number {
    return (nodes[index] ?? oob()).parentIndex;
}
```

and calling it in the five places.